### PR TITLE
Skip validation if we just remove the finalizer

### PR DESCRIPTION
**Pull request by Fedosin**
To allow users to delete their machines with invalid specs, we have to allow them to remove finalizers. In this case all other validations will be skipped even if machine's spec is incorrect.

### CODE REVIEW

File: pkg/webhooks/machine_webhook.go [hunk 1]
```diff
@@ -487,6 +488,12 @@ func MachineSetMutatingWebhook() admissionregistrationv1.MutatingWebhook {
 }
 
 func (h *machineValidatorHandler) validateMachine(m, oldM *machinev1.Machine) (bool, []string, utilerrors.Aggregate) {
+	// Skip validation if we just remove the finalizer.
```
Comment (JoelSpeed) [inline pkg/webhooks/machine_webhook.go:491]: You need to check that everything else is the same no? This change allows any change when removing the finalizer right?

File: pkg/webhooks/machine_webhook.go [hunk 2]
```diff
@@ -1426,3 +1438,14 @@ func validateAzureDataDisks(machineName string, spec *machinev1.AzureMachineProv
...
+
+// isFinalizerOnlyRemoval checks if the machine update only removes finalizers.
+func isFinalizerOnlyRemoval(m, oldM *machinev1.Machine) (bool, error) {
+	patchBase := client.MergeFrom(oldM)
+	data, err := patchBase.Data(m)
+	if err != nil {
+		return false, fmt.Errorf("cannot calculate patch data from machine object: %w", err)
+	}
+
+	return string(data) == `{"metadata":{"finalizers":[""]}}`, nil
```
Comment (JoelSpeed) [inline pkg/webhooks/machine_webhook.go:1450]: Still an open question about what happens if there are multiple finalizers, can we do something to make sure that controllers can remove individual finalizers without removing all the finalizers?

Comment (Fedosin) [reply pkg/webhooks/machine_webhook.go:1450]: Could you describe a use case when we need to delete individual finalizers on invalid machines. I think the goal is to allow users to delete machines that were accidentally created with invalid specs. To do so we always need to remove all finalizers, right?

Comment (JoelSpeed) [reply pkg/webhooks/machine_webhook.go:1450]: I had thought you could only remove one finalizer at a time, but this seems to not be the case, happy to leave as is

**Pull request merged**